Marlin/Marlin_main.cpp

```diff
@@ -5461,7 +5516,13 @@ void home_all_axes() { gcode_G28(true); }
               #if ENABLED(PROBE_MANUALLY)
                 z_at_pt[axis] += lcd_probe_pt(cos(a) * r, sin(a) * r);
               #else
-                z_at_pt[axis] += probe_pt(cos(a) * r + dx, sin(a) * r + dy, stow_after_each, 1, false);
+                z_at_pt[axis] += probe_pt(cos(a) * r + dx, sin(a) * r + dy, stow_after_each, 1);
+                if (isnan(z_at_pt[axis])) {
+                  LCD_MESSAGEPGM(MSG_ERR_PROBING_FAILED);
+                  SERIAL_ERROR_START();
+                  SERIAL_ERRORLNPGM(MSG_ERR_PROBING_FAILED);
```

Comment:
While I 100% agree, I wasn't really sure if I wanted to try and tackle refactoring this function as part of this PR. It's a very complicated function that I don't 100% understand yet. The main issue is that there is some setup at the beginning of the function and some cleanup that needs to occur before the function ends, and I wanted to make sure that the cleanup happened if the function was going to bail because of a failed probe attempt.

Regarding the repetition of those three lines, I just looked elsewhere to see how displaying something in the terminal and on the LCD was handled, and that's what I found. A helper function/macro that displays the same message in both places could certainly help with the duplication. We could call it something like `LCD_SERIAL_MESSAGEPGM`. Work for you?